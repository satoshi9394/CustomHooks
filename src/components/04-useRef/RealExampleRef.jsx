import React, { useState } from 'react'
import MultipleCustomHooks from '../03-example/MultipleCustomHooks';


import './effectStyles.css';

const RealExampleRef = () => {

  const [show, setShow] = useState(false)

  return (
    <>
      <h1>
        RealExampleRef
      </h1>
      <hr/>
      {
        show && <MultipleCustomHooks/>
      }
      <button
        className="btn btn-success mt-1"
        onClick={ () => { setShow(!show) }}
      >
        Mostrar/Ocultar
      </button>
    </>
  )
}

export default RealExampleRef
