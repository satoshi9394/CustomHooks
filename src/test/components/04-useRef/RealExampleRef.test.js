import { shallow } from 'enzyme';
import React from 'react';
import RealExampleRef from '../../../components/04-useRef/RealExampleRef';


describe('Pruebas del componente <RealExampleRef/>', () => {

  const wrapper = shallow( <RealExampleRef/>);

  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
  });
  
  test('debe mostrarse el compoenente de <MultipleCustomHooks', () => {
    wrapper.find('button').simulate('click');
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
  })

})