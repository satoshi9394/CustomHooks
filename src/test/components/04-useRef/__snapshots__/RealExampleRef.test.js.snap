// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <RealExampleRef/> debe mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-success mt-1"
    onClick={[Function]}
  >
    Mostrar/Ocultar
  </button>
</Fragment>
`;
